pydantic/_internal/_decorators.py

```diff
@@ -264,15 +265,37 @@ def __init__(
         self.func = func
         self.unwrapped_func = unwrapped_func
         self.info = info
+        self.shim = shim
 
-
-AnyDecorator = Union[
-    Decorator[ValidatorDecoratorInfo],
-    Decorator[FieldValidatorDecoratorInfo],
-    Decorator[RootValidatorDecoratorInfo],
-    Decorator[FieldSerializerDecoratorInfo],
-    Decorator[ModelSerializerDecoratorInfo],
-]
+    @staticmethod
+    def build(
+        cls_: Any,
+        cls_var_name: str,
+        shim: Callable[[Any], Any] | None,
+        info: DecoratorInfoType,
+    ) -> Decorator[DecoratorInfoType]:
+        val = getattr(cls_, cls_var_name)
+        if isinstance(val, PydanticDecoratorMarker):
+            val = val.wrapped
```

Comment:
This allows the child to override without using the decorator again. But maybe we want to force users to explicitly apply the decorator?